### PR TITLE
[style] Dev prefix unexported globals with  

### LINKED ISSUE #2946: [style] prefix unexported globals with '_'
https://github.com/uber-go/guide/blob/master/style.md#prefix-unexported-globals-with-_

### Prefix Unexported Globals with _

Prefix unexported top-level `var`s and `const`s with `_` to make it clear when
they are used that they are global symbols.

Exception: Unexported error values, which should be prefixed with `err`.

Rationale: Top-level variables and constants have a package scope. Using a
generic name makes it easy to accidentally use the wrong value in a different
file.

<table>
<thead><tr><th>Bad</th><th>Good</th></tr></thead>
<tbody>
<tr><td>

```go
// foo.go

const (
  defaultPort = 8080
  defaultUser = "user"
)

// bar.go

func Bar() {
  defaultPort := 9090
  ...
  fmt.Println("Default port", defaultPort)

  // We will not see a compile error if the first line of
  // Bar() is deleted.
}
```

</td><td>

```go
// foo.go

const (
  _defaultPort = 8080
  _defaultUser = "user"
)
```

</td></tr>
</tbody></table>

**Exception**: Unexported error values may use the prefix `err` without the underscore.

### PR DESCRIPTION
**Pull request by LuckyPigeon**
# Description

prefix all unexported top-level `var`s and `const`s with "_". Make them be cleared when they use as global variable.

Fixes #2946 

## Type of change

Please delete options that are not relevant.

* Bug fix (non-breaking change which fixes an issue)
* Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

* `go.test.sh`

**Test Configuration**:

- Firmware version: Windows 10/Ubuntu 20.04 
- Hardware:
- Toolchain:
- SDK:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by LuckyPigeon** [comment]
@Liuhaai 
This PR is too big, I'll open for others separately.

File: action/protocol/execution/evm/evmstatedbadapter_test.go [hunk 1]
```diff
@@ -90,7 +90,7 @@ func TestAddBalance(t *testing.T) {
 
 	sm, err := initMockStateManager(ctrl)
 	require.NoError(err)
-	addr := common.HexToAddress("02ae2a956d21e8d481c3a69e146633470cf625ec")
+	addr := common.HexToAddress("02ae2a956d21e8d481_c3a69e146633470cf625ec")
```
Comment (Liuhaai) [inline action/protocol/execution/evm/evmstatedbadapter_test.go:93]: revert

File: api/coreservice_test.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	// "context"
+	"testing"
+	// "github.com/golang/mock/gomock"
+	// "github.com/iotexproject/iotex-proto/golang/iotexapi"
+	// "github.com/pkg/errors"
+	// "github.com/stretchr/testify/require"
+	// "github.com/iotexproject/iotex-core/test/mock/mock_apicoreservice"
+	// "github.com/iotexproject/iotex-core/testutil"
+)
+
```
Comment (Liuhaai) [inline api/coreservice_test.go:19]: revert the change on this file.

**Review by Liuhaai — APPROVED** [review]

**Comment by Liuhaai** [comment]
Much appreciate the hard work

**Review by CoderZhi — APPROVED** [review]
